help/default.js

```diff
@@ -0,0 +1,267 @@
+        return;
+    }
+    event.preventDefault();
+    document.getElementById('search-submit').click();
+});
+
+// Perform search when search button is clicked
+// Assuming search-submit is defined
+document.getElementById('search-submit').addEventListener('click', function () {
+    const query = g_searchInput.value.trim().toLowerCase();
+    g_searchResults.innerHTML = '';
+    g_currentSelectionIndex = -1;
+
+    if (query.length === 0) {
+        g_searchResults.style.display = 'none';
+        return;
+    }
+
+    if (g_index === null) {
+        // Assuming the fetch works correctly without console.log
+        fetch('/searchIndex.json')
+            .then((response) => response.json())
+            .then((indexData) => {
+                g_index = new window.FlexSearch.Document({
+                    document: {
+                        id: 'id',
+                        index: ['content'],
+                        store: ['title', 'url', 'content'],
+                    },
+                });
+
+                for (const [key, data] of Object.entries(indexData)) {
+                    g_index.import(key, data);
+                }
+
+                performSearch(query);
+            });
+    } else {
+        performSearch(query);
+    }
+});
+
+function performSearch(query) {
+    const results = g_index.search({
+        query,
+        field: 'content',
+    });
+
+    if (results && results.length > 0) {
+        g_searchResultsArray = [];
+        results.forEach((result) => {
+            result.result.forEach((docId) => {
+                const doc = g_index.store[docId];
+                if (doc && doc.content) {
+                    const searchTermIndex = doc.content.toLowerCase().indexOf(query);
+                    const contextBefore = doc.content.substring(Math.max(0, searchTermIndex - 30), searchTermIndex);
+                    const contextAfter = doc.content.substring(searchTermIndex + query.length, Math.min(doc.content.length, searchTermIndex + query.length + 30));
+                    const searchResultHtml = `
+                        <div class="search-result">
+                            <a href="${doc.url}" class="search-result-title">${doc.title}</a>
+                            <div class="search-result-context">...${contextBefore}<strong>${query}</strong>${contextAfter}...</div>
+                        </div>
+                    `;
+                    const resultElement = document.createElement('div');
+                    resultElement.innerHTML = searchResultHtml;
+                    g_searchResults.appendChild(resultElement);
+                    g_searchResultsArray.push(resultElement);
+
+                    if (g_searchResultsArray.length === 1) {
+                        g_currentSelectionIndex = 0;
+                        updateSelectedResult();
+                    }
+                }
+            });
+        });
+        g_searchResults.style.display = 'block';
```

Comment:
Styles should only be applied by classes or IDs. I suggest having some classes like `display-block` and `display-none`